### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by Natia-bit**
Based on feedback changed the code. 

deleteMeal:  I added the second delete after docker. I think there is a permission issue with delete and my custom version of docker image. When I run it locally it worked perfectly with single delete, but when I use it via docker container it did now complete delete giving me an error based on frequency and foreign key. 
I did try changing permission on docker file, but that didn't work either

### CODE REVIEW

File: src/main/java/MealDemo/entity/Frequency.java [hunk 1]
```diff
@@ -8,7 +8,7 @@ public class Frequency {
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     @Column(name="id")
-    private int id;
+    private long id;
```
Comment (charles-van-wyk) [inline src/main/java/MealDemo/entity/Frequency.java:11]: Not related to the PR but something to add to your todo list: What is the difference between `long` and `Long` or `int` and `Integer`?

File: src/main/java/MealDemo/service/MealPlanningServiceImpl.java [hunk 2]
```diff
@@ -35,109 +43,131 @@ public List<Frequency> getFrequencies() {
...
         tempFreq.setMeal(meal);
         frequencyRepository.save(tempFreq);
         System.out.println("meal saved! ");
     }
 
+
+    @Transactional
     @Override
     public boolean deleteMeal(int mealId) {
+        boolean isDeleted = false;
```
Comment (charles-van-wyk) [inline src/main/java/MealDemo/service/MealPlanningServiceImpl.java:71]:
```suggestion
        var isDeleted = false;
```

File: src/main/java/MealDemo/service/MealPlanningServiceImpl.java [hunk 3]
```diff
@@ -35,109 +43,131 @@ public List<Frequency> getFrequencies() {
...
-
-        existingMeal.setMealName(meal.getMealName());
-        existingMeal.setCategory(meal.getCategory());
-
-        mealRepository.save(existingMeal);
-
-        System.out.println("meal has been updated. Meal details \n" + existingMeal.toString());
+        var tempMeal = mealRepository.findById(mealId);
+        if (tempMeal.isPresent()) {
+            var existingMeal = mealRepository.getById(mealId);
```
Comment (charles-van-wyk) [inline src/main/java/MealDemo/service/MealPlanningServiceImpl.java:92]: You already have the `tempMeal` value here, you don't need to read it from the database again. You can simple use `var existingMeal = tempMeal.get()`

File: src/main/java/MealDemo/service/MealPlanningServiceImpl.java [hunk 4]
```diff
@@ -35,109 +43,131 @@ public List<Frequency> getFrequencies() {
...
     }
 
 
     @Override
     public Map<String, List<Meal>> mealsByCategories() {
-        List<Meal> allMeals = getMeals();
+        var allMeals = getMeals();
         Map<String, List<Meal>> mealsByCategoriesList = new HashMap<>();
-        for (Meal meal : allMeals) {
+        for (var meal : allMeals) {
```
Comment (charles-van-wyk) [inline src/main/java/MealDemo/service/MealPlanningServiceImpl.java:107]: Neat, I didn't know the `var` keyword was usable in loops too... Now I learned something too

File: src/main/java/MealDemo/service/MealPlanningServiceImpl.java (hunk 3 above)
Comment (Natia-bit) [reply src/main/java/MealDemo/service/MealPlanningServiceImpl.java:92]: That was obvious, I don't know why I didn't do that

**Pull request merged**